gateway/handlers/createhandler.go

```diff
@@ -66,7 +77,7 @@ func MakeNewFunctionHandler(metricsOptions metrics.MetricOptions, c *client.Clie
 	}
 }
 
-func makeSpec(request *requests.CreateFunctionRequest, maxRestarts uint64, restartDelay time.Duration) swarm.ServiceSpec {
+func makeSpec(request *requests.CreateFunctionRequest, maxRestarts uint64, restartDelay time.Duration, secrets []*swarm.SecretReference) swarm.ServiceSpec {
```

Comment:
@LucasRoesler this must work for create and update deployment before it can be merged.

To update via CLI:

```
faas-cli update --replace=false --update=true
```

Reply:
I am going ot push a change that fixes the `updateSpec` any opinion on if I should move the `makeSecretsArray` method to be in a central location?  It is currently in the `createhandler.go` file but will also be used in `update_handler.go`